### PR TITLE
Reenable skipped tests health check

### PR DESCRIPTION
**Pull request by fredo**
## Remove concept of broadcast rooms
This PR removes broadly the concept of having broadcast rooms. Switching to presenceless Raiden, broadcast rooms are not used anymore but functions to join and monitor remained in the code base unused. Now, almost every function and constant regarding broadcast rooms is removed. The rest should follow in another PR. Untouched is `GMatrixClient` and the Broadcast rooms prefixes as they still might be used by other projects such as the services.

## Reenabling skipped tests
A lot of tests were skipped with the reason `Test is still using presence / health check`. Most of the tests used either health check functions or some `wait_for_[reachability]` which used presence internally. This was necessary as Raiden were not sending messages to offline nodes introducing races between the presence status and sending a message. With the new version, the transport layer will not check anymore if another node is online. Thus, removing these checks was sufficient for most of the tests. 

## Removed tests
Other tests could be removed as they were testing removed code. Especially when it comes to broadcast rooms. Other tests, testing presence updates, should be removed and moved to the PFS codebase which is handling all about presences now.

List of removed tests: 

- test_matrix_tx_error_handling
- test_join_invalid_discovery
- test_matrix_discovery_room_offline_server
- test_reproduce_handle_invite_send_race_issue_3588
- test_transport_does_not_receive_broadcast_rooms_updates
- test_transport_presence_updates **should be moved to raiden services**
- test_transport_capabilities **should be moved to raiden services**
- test_assumption_search_user_directory_returns_federated_users
- test_assumption_federation_works_after_original_server_goes_down
- test_admin_is_allowed_to_kick

### CODE REVIEW

File: raiden/tests/scenarios/bf6_stress_hub_node.yaml [hunk 1]
```diff
@@ -52,7 +52,7 @@ nodes:
     8:
       matrix-server: {{ matrix_servers[3] }}
     9:
-      matrix-server: {{ matrix_servers[4] }}
+      matrix-server: {{ matrix_servers[3] }}
```
Comment (netcriptus) [inline raiden/tests/scenarios/bf6_stress_hub_node.yaml:55]: I'm not sure I understood why this change is relevant for the tests

**Review by netcriptus — APPROVED** [review]
Overall LGTM, just one change wasn't clear to me.

File: raiden/tests/scenarios/bf6_stress_hub_node.yaml (hunk 1 above)
Comment (fredo) [reply raiden/tests/scenarios/bf6_stress_hub_node.yaml:55]: It's not. It's actually reverting an older commit when we skipped `transport05` because it was not working properly. Now the scenario uses all transport servers again like it used to do.
